app_dart/lib/src/model/ci_yaml/target.dart

```diff
@@ -40,12 +40,32 @@ class Target {
 
   static String kIgnoreFlakiness = 'ignore_flakiness';
 
-  /// Gets dimensions for this [pb.Target].
+  /// Gets assembled dimensions for this [pb.Target].
   ///
   /// Swarming dimension doc: https://chromium.googlesource.com/infra/luci/luci-go/+/HEAD/lucicfg/doc/README.md#swarming.dimension
+  ///
+  /// Target dimensions are prioritized in:
+  ///   1. [schedulerConfig.platformDimensions]
+  ///   2. [pb.Target.properties]
   List<RequestedDimension> getDimensions() {
-    final Map<String, Object> properties = getProperties();
     final List<RequestedDimension> dimensions = <RequestedDimension>[];
+
+    final Map<String, Object> platformDimensions = _getPlatformDimensions();
+    for (String key in platformDimensions.keys) {
+      String value = platformDimensions[key].toString();
+      dimensions.add(RequestedDimension(key: key, value: value));
+    }
+
+    final Map<String, Object> targetDimensions = _getTargetDimensions();
+    for (String key in targetDimensions.keys) {
+      String value = targetDimensions[key].toString();
+      dimensions.add(RequestedDimension(key: key, value: value));
```

Comment:
This doesn't handle the case where the target overrides the platform dimension.

If platform's cpu is x86, and the target wants arm64 it'll request both and cause an infra failure